fix: TS2532 in inventory-move-item-dialog.test.ts:806 blocking all PR CI

## Problem

PR #919 introduced a TypeScript strict-mode error that is now blocking CI on all open PRs:

```
web-components/inventory-move-item-dialog.test.ts(806,16): error TS2532: Object is possibly 'undefined'.
```

**File:** `static/js/web-components/inventory-move-item-dialog.test.ts`
**Line 806:** `expect(el.searchResults[0].identifier).to.equal('toolbox_garage');`

## Root Cause

TypeScript strict mode flags `array[0]` as `T | undefined` — accessing `.identifier` directly is not safe without a non-null assertion or a prior existence check.

## Fix

Change line 806 (and any similar patterns in the same file) to use either:

Option A — non-null assertion:
```ts
expect(el.searchResults[0]!.identifier).to.equal('toolbox_garage');
```

Option B — local variable with prior assertion:
```ts
const first = el.searchResults[0];
expect(first).to.exist;
expect(first!.identifier).to.equal('toolbox_garage');
```

Scan the entire file for any similar patterns (`searchResults[N].`, `el.\w+\[0\].`) and fix them all in the same PR.

This is urgent — it is currently blocking every open PR in the repo from passing CI.

--- a/static/js/web-components/inventory-move-item-dialog.test.ts
+++ b/static/js/web-components/inventory-move-item-dialog.test.ts
@@ -803,7 +803,7 @@ describe('InventoryMoveItemDialog', () => {
       });
 
       it('should filter out current container', () => {
-        expect(el.searchResults[0].identifier).to.equal('toolbox_garage');
+        expect(el.searchResults[0]!.identifier).to.equal('toolbox_garage');
       });
 
       it('should set searchLoading to false', () => {

